geode-core/src/integrationTest/java/org/apache/geode/cache/query/internal/ExecutionContextIntegrationTest.java

```diff
@@ -0,0 +1,340 @@
+              "No such iterator with name = " + rItr.getName() + " should be available");
+      }
+    }
+  }
+
+  private void assertIteratorScopeMultiThreaded(Iterator itr) {
+    RuntimeIterator rItr = (RuntimeIterator) itr.next();
+    switch (rItr.getName()) {
+      case "p":
+        assertThat(rItr.getScopeID())
+            .as("The scopeID of outer iterator is not 1").isEqualTo(1);
+        break;
+      case "pf":
+      case "pos":
+        assertThat(rItr.getScopeID())
+            .as("The scopeID of first inner level iterator is not 2").isEqualTo(2);
+        break;
+      case "rtPos":
+        assertThat(rItr.getScopeID())
+            .as("The scopeID of second inner level iterator is not 3").isEqualTo(3);
+        break;
+      case "y":
+        assertThat(rItr.getScopeID())
+            .as("The scopeID of outer level iterator is not 1").isEqualTo(1);
+        break;
+      default:
+        throw new RuntimeException(
+            "No such iterator with name = " + rItr.getName() + " should be available");
+    }
+  }
+
+  private int computeEvaluateAndAssertIterator(ExecutionContext context, int i,
+      CompiledIteratorDef iterDef) throws TypeMismatchException, NameResolutionException {
+    ++i;
+    @SuppressWarnings("unchecked")
+    Set<RuntimeIterator> dependencies = iterDef.computeDependencies(context);
+    context.addDependencies(new CompiledID("dummy"), dependencies);
+    RuntimeIterator rIter = iterDef.getRuntimeIterator(context);
+    context.addToIndependentRuntimeItrMap(iterDef);
+    context.bindIterator(rIter);
+
+    assertThat(rIter.getIndexInternalID())
+        .as("The index_internal_id is not set as per expectation of index_iter'n'")
+        .isEqualTo("index_iter" + i);
+
+    return i;
+  }
+
+  @Test
+  public void constructorShouldUseConfiguredMethodAuthorizer() {
+    ExecutionContext unsecuredContext = new QueryExecutionContext(null, server.getCache());
+    MethodInvocationAuthorizer noOpAuthorizer = unsecuredContext.getMethodInvocationAuthorizer();
+
+    // No security, no-op authorizer.
+    assertThat(noOpAuthorizer).isNotNull();
+    assertThat(noOpAuthorizer.getClass().getCanonicalName())
+        .startsWith("org.apache.geode.cache.query.internal.DefaultQueryService$$Lambda$");
+
+    server.stopMember();
+
+    // Security Enabled -> RestrictedMethodAuthorizer
+    server.withProperty(SECURITY_MANAGER, SimpleSecurityManager.class.getName())
+        .withProperty("security-username", "cluster")
+        .withProperty("security-password", "cluster")
+        .startServer();
+    ExecutionContext securedContext = new QueryExecutionContext(null, server.getCache());
+    MethodInvocationAuthorizer authorizer = securedContext.getMethodInvocationAuthorizer();
+    assertThat(authorizer).isNotNull();
+    assertThat(authorizer).isInstanceOf(RestrictedMethodAuthorizer.class);
+  }
+
+  @Test
+  public void testFunctionalAddToIndependentRuntimeItrMapWithoutIndex() throws Exception {
```

Comment:
what is this testing (more descriptive name?)  It looks like we are checking that the internal id is the same as the index internal id?